crates/daphne-server/tests/e2e/e2e.rs

```diff
@@ -1348,6 +1496,116 @@ async fn leader_selected() {
     .unwrap();
 }
 
+#[tokio::test]
+async fn leader_collect_taskprov_repeated_abort() {
```

Comment:
What does this new test do?

Reply:
It checks that the Leader aborts if the `taskprov` extension appears in both the public and private extensions.

Reply:
I fixed a bug in it too.

Reply:
Can we move this to the unit tests in the `daphne` crate. or does exercise a behavior that's not implemented by `InMemoryAggregator`?